### PR TITLE
Update to webpack 5 to enable code splitting

### PR DESCRIPTION
**Pull request by TheAncientGoat**
Thanks for the great crate!

I wanted to render an app that has a few crazy dependencies that v8 chokes on; this PR addresses the issue by enabling code splitting - preventing the problematic libraries from being parsed during SSR. 

In order to do so, I had to change to the esmodule output format to prevent webpack from referring to a global window object. This unfortunately breaks react-scripts compatability, so I included Parcel as a quick runner (a custom dev webpack script is probably a better solution; as parcel doesn't build working a working SSR target - can't figure out how to wrap the output in a var like webpack does)

I also noticed some strange behavior with streaming responses in actix - if you refresh rapidly sometimes the react app fails to hydrate. I added some handling to check whether the dom node is available and to only attempt once that is the case, but it still occurred frequently enough to just use a plain body response.

This also includes an update to the latest actix beta that I tried to debug the streaming problem; the problem persisted, but might as well keep up to date to be ready for the release

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -2,36 +2,43 @@
...
   },
   "eslintConfig": {
     "extends": [
       "react-app",
       "react-app/jest"
     ]
   },
+  "targets": {
+    "client": {
+      "source": "src/index.html"
```
Comment (Valerioageno) [inline client/package.json:36]: Should we create this file as well?

**Review by Valerioageno — CHANGES_REQUESTED** [review]
Hey Ryan, thanks for the updates! 
I've opened a branch for this purpose long time ago but I never finish it.
Well done! I leave to you a comment and please update the `.gitignore` file with the parcel artifacts.
In order to overcome the actix bug I have some ideas about store the evaluated JS inside a struct avoiding call the `render_to_string()` function on each request but just on the first run.

**Comment by Valerioageno** [comment]
Please also update the `client/README.md` about the changes made from the create-react-app scaffolding in order to keep track of them

**Comment by TheAncientGoat** [comment]
Thanks for the feedback - will address the changes; storing the render result in some form of cache is a good idea, but unfortunately I need fresh renders based on props so I don't think it's a universal solution

**Comment by Valerioageno** [comment]
Well actually the idea is not to store the render result which, as we expect, change at each request, but keep in memory the context, probably using `v8::Global<v8::Context>>` in the struct. 
Currently at **each** request calling the `render_to_string()` method we also perform a new v8 platform initialization.

Inject custom params is performed just in the latest part of the function:

```rust
for key in fn_map.keys() {
      let result = fn_map[key]
            .call(scope, undef, &[params])
            .expect("Are provided all needed props?");

      let result = result.to_string(scope).unwrap();

      rendered = format!("{}{}", rendered, result.to_rust_string_lossy(scope));
}
```
For this reason if we store the previous part in memory and just call this at each request it'll be more performant.
Here a repo where I tried to start with a full project using react and actix. [Reactix](https://github.com/Valerioageno/reactix).
Until this update is not merged performance are not comparable with competitor runtimes.

File: client/package.json (hunk 1 above)
Comment (TheAncientGoat) [reply client/package.json:45]: Added!

**Pull request merged**